tokio/src/runtime/scheduler/multi_thread/queue.rs

```diff
@@ -10,6 +9,22 @@ use std::mem::MaybeUninit;
 use std::ptr;
 use std::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed, Release};
 
+// Use wider integers when possible to increase ABA resilience.
+//
+// See issue #5041: <https://github.com/tokio-rs/tokio/issues/5041>.
+cfg_has_atomic_u64! {
+    type UnsignedShort = u32;
+    type UnsignedLong = u64;
+    type AtomicUnsignedShort = crate::loom::sync::atomic::AtomicU32;
+    type AtomicUnsignedLong = crate::loom::sync::atomic::AtomicU64;
+}
+cfg_not_has_atomic_u64! {
+    type UnsignedShort = u16;
+    type UnsignedLong = u32;
+    type AtomicUnsignedShort = crate::loom::sync::atomic::AtomicU16;
+    type AtomicUnsignedLong = crate::loom::sync::atomic::AtomicU32;
+}
```

Comment:
Unfortunately, the `BITS` constant is not available on the minimum Rust version we support.
```
error[E0658]: use of unstable library feature 'int_bits_const'
   --> tokio/src/runtime/scheduler/multi_thread/queue.rs:526:58
    |
526 |     (real as UnsignedLong) | ((steal as UnsignedLong) << UnsignedShort::BITS)
    |                                                          ^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #76904 <https://github.com/rust-lang/rust/issues/76904> for more information
```
You will need to define the constant yourself:
```suggestion
cfg_has_atomic_u64! {
    type UnsignedShort = u32;
    const SHORT_BITS: u32 = 32;
    type UnsignedLong = u64;
    type AtomicUnsignedShort = crate::loom::sync::atomic::AtomicU32;
    type AtomicUnsignedLong = crate::loom::sync::atomic::AtomicU64;
}
cfg_not_has_atomic_u64! {
    type UnsignedShort = u16;
    const SHORT_BITS: u32 = 16;
    type UnsignedLong = u32;
    type AtomicUnsignedShort = crate::loom::sync::atomic::AtomicU16;
    type AtomicUnsignedLong = crate::loom::sync::atomic::AtomicU32;
}
```

Reply:
I'm fine either way. What you did is also fine.